frame/vamm/src/lib.rs

```diff
@@ -0,0 +1,332 @@
+//!
+//! ## Overview
+//!
+//! ![](http://www.plantuml.com/plantuml/svg/NSqz2W91343XtbFe0TpqLWk2zyXcuyv09XdoWzTNB2oi7b_rraZqh26dIrUIshbSpYrpnWt0yRKSFLjj5Unacgova0susvWMk0a_Ej0Fm46D7PwEWu64qRiUrsOL_roce7xFA-l-wHi0)
+//!
+//! TODO
+//!
+//! ### Terminology
+//!
+//! * **VAAM**: Acronym for Virtual Automated Market Maker. (TODO: expand definition)
+//!
+//! TODO
+//!
+//! ### Goals
+//!
+//! TODO
+//!
+//! ### Actors
+//!
+//! TODO
+//!
+//! (clearing house)
+//!
+//! ### Implementations
+//!
+//! The VAMM pallet provides implementations for the following traits:
+//!
+//! - [`Vamm`](composable_traits::vamm::Vamm): Exposes functionality for
+//! creating, managing and deprecating virtual automated market makers.
+//!
+//! ## Interface
+//!
+//! TODO
+//!
+//! ### Extrinsics
+//!
+//! TODO
+//!
+//! not applicable
+//!
+//! ### Implemented Functions
+//!
+//! * [`create`](<Pallet as Vamm>::create)
+//!
+//! TODO
+//!
+//! ### Public Functions
+//!
+//! TODO
+//!
+//! ### Public Storage Objects
+//!
+//! TODO
+//!
+//! ## Usage
+//!
+//! TODO
+//!
+//! ### Example
+//!
+//! TODO
+//!
+//! ## Related Modules
+//!
+//! TODO
+//!
+//! (clearing house)
+//!
+//! <!-- Original author: @Cardosaum -->
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(test)]
+mod mock;
+
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+	// ----------------------------------------------------------------------------------------------------
+	//                                       Imports and Dependencies
+	// ----------------------------------------------------------------------------------------------------
+
+	use codec::{Codec, FullCodec};
+	use composable_traits::vamm::*;
+	use frame_support::{pallet_prelude::*, sp_std::fmt::Debug, transactional, Blake2_128Concat};
+	use sp_runtime::{
+		traits::{AtLeast32BitUnsigned, CheckedAdd, CheckedDiv, CheckedMul, CheckedSub, One, Zero},
+		ArithmeticError,
+	};
+
+	// ----------------------------------------------------------------------------------------------------
+	//                                    Declaration Of The Pallet Type
+	// ----------------------------------------------------------------------------------------------------
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	// ----------------------------------------------------------------------------------------------------
+	//                                             Config Trait
+	// ----------------------------------------------------------------------------------------------------
+
+	// Configure the pallet by specifying the parameters and types on which it depends.
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// Event type emitted by this pallet. Depends on the runtime's definition of an event.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		/// The `VammId` used by the pallet. Corresponds to the Ids used by the Vamm pallet.
```

Comment:
This *is* the Vamm pallet, no?

Reply:
Yep, might need to write more clearly. The comment seemed a bit off.

Reply:
What about the following one?

```
/// The Ids used by the pallet to index each virtual automated market maker created.
```

cc/ @0xangelo

Reply:
Looks great 👌